server/store/user_store.go

```diff
@@ -230,6 +239,26 @@ func (s *pluginStore) DeleteUserFromIndex(mattermostUserID string) error {
 	})
 }
 
+func (s *pluginStore) SearchInUserIndex(term string, limit int) ([]UserShort, error) {
+	userIndex, err := s.LoadUserIndex()
+	if err != nil {
+		return nil, errors.Wrap(err, "error searching user in index")
+	}
+
+	result := make([]UserShort, 0)
+	for idx, u := range userIndex {
+		if strings.Contains(u.MattermostUsername, term) || strings.Contains(u.MattermostDisplayName, term) || strings.Contains(u.Email, term) {
```

Comment:
I wonder if the user would want to be able to invite arbitrary emails too. I suppose we can still let the user can do that, even the input doesn't match a connected user. The UI could just allow them to put whatever email they want.

@matthewbirtch Do you have any thoughts on this? We're implementing the "Attendees" selector in the create event modal. At the moment, the autocomplete searches for connected users, and find matches by comparing the autocomplete input with a user's username, display name in Mattermost, and connected email address. If any of them match, the user shows in the selector.

I would normally suggest we just use the Mattermost user autocomplete for this kind of thing, but I think we want to limit the results to connected users so we know their calendar email address. We could even try and use gcal's API for this autocomplete, but that is more complex and requires rework for compatibility with mscalendar.

The Figma design currently shows the Mattermost display name in the selected options. Would the autocomplete options be shown the same way? 0/5 I'm thinking the email address is significant here, since that's what is used on the calendar side.

<img width="1044" alt="CleanShot 2023-07-28 at 19 13 19@2x" src="https://github.com/mattermost/mattermost-plugin-mscalendar/assets/6913320/a3252b53-fac6-4628-90fd-69c435d39c88">

Reply:
Yeah that's why I made the code find by everything, but we only have emails (thus, can use as attendees) of connected users. But I think that we can let it use arbitrary emails too. I will make some tests of that next week.

Reply:
I agree, that would be great to be able to invite arbitrary email addresses in addition to connected users. We do this in the 'Invite people to team' modal right now by adding an option in the autocomplete to invite via email

Autocomplete displays like this:
<img width="648" alt="image" src="https://github.com/mattermost/mattermost-plugin-mscalendar/assets/2040554/4ac04436-8220-4623-81b9-bddfb5de7d5d">

Chips display like this
<img width="659" alt="image" src="https://github.com/mattermost/mattermost-plugin-mscalendar/assets/2040554/9a15dc1f-f915-4189-9e8b-7e43137351ab">

Is it possible to leverage this behavior?